docs/Building SpiderMonkey.md

```diff
@@ -0,0 +1,62 @@
+# Building SpiderMonkey for Embedders #
+
+Use these instructions to build your own copy of SpiderMonkey.
+
+## Prerequisites ##
+
+You will need a **C++ compiler** that can handle the C++14 standard,
+**GNU Make**, **zlib**, and **libffi**.
+These can usually be installed with a package manager.
+You will also need **ICU** of at least version 59.1.
+ICU can also usually be installed with a package manager, but the
+version might not be new enough.
+
+> **NOTE** If you can't install a new enough version of ICU, you can
+> build the copy bundled with SpiderMonkey by omitting
+> `--with-system-icu` in the build instructions below, at the cost of a
+> longer build time.
+
+## Getting the source code ##
+
+Currently, the most reliable way to get the SpiderMonkey source code is
+to download the Firefox source.
+At the time of writing, the latest source for Firefox ESR 60, which
+contains the source for SpiderMonkey ESR 60, can be found here:
+http://ftp.mozilla.org/pub/firefox/releases/60.3.0esr/source/
+
+> **NOTE** Mozilla may be able to provide separate source packages for
+> SpiderMonkey in the future, but this is difficult for a number of
+> reasons.
+
+## Building SpiderMonkey ##
+
+First you should decide where you want to install SpiderMonkey.
+By default, it will install into `/usr/local`.
+You might want to pick some other location if `/usr/local` is not
+writable to you without superuser permissions, for example.
+
+```sh
+cd js/src
+mkdir _build
+cd _build
+../configure --disable-jemalloc  --enable-posix-nspr-emulation \
+    --with-system-zlib --with-system-icu --with-intl-api \
+    --enable-debug --enable-optimize AUTOCONF=autoconf
```

Comment:
I think autoconf-2.13 is generally available as package and it might be easier to just suggest they have it installed. I also selfishly suggest this because it is closing to being able to contribute to spidermonkey itself.

Reply:
Yeah, I can see the point of that. It's kind of annoying for people who just want to build SpiderMonkey, though. The GNOME platform people balked at including it in the platform build, for example.

I can't find a way to mention the `AUTOCONF=autoconf` workaround without making it too attractive to just take that shortcut, though, so I'll leave it out.